h5t_types_test.go

```diff
@@ -4,35 +4,50 @@
 
 package hdf5
 
-import (
-	"runtime"
-	"testing"
-	"time"
-)
+import "testing"
 
 func TestSimpleDatatypes(t *testing.T) {
 	// Smoke tests for the simple datatypes
-	tests := []interface{}{
-		int(0),
-		int8(0),
-		int16(0),
-		int32(0),
-		int64(0),
-		uint(0),
-		uint8(0),
-		uint16(0),
-		uint32(0),
-		uint64(0),
-		float32(0),
-		float64(0),
-		string(""),
-		bool(true),
+	tests := []struct {
+		v          interface{}
+		illegalPtr bool
+	}{
+		{v: int(0), illegalPtr: false},
+		{v: int8(0), illegalPtr: false},
+		{v: int16(0), illegalPtr: false},
+		{v: int32(0), illegalPtr: false},
+		{v: int64(0), illegalPtr: false},
+		{v: uint(0), illegalPtr: false},
+		{v: uint8(0), illegalPtr: false},
+		{v: uint16(0), illegalPtr: false},
+		{v: uint32(0), illegalPtr: false},
+		{v: uint64(0), illegalPtr: false},
+		{v: float32(0), illegalPtr: false},
+		{v: float64(0), illegalPtr: false},
+		{v: string(""), illegalPtr: false},
+		{v: bool(true), illegalPtr: false},
+		{v: (*int)(nil), illegalPtr: false},
+		{v: (*int8)(nil), illegalPtr: false},
+		{v: (*int16)(nil), illegalPtr: false},
+		{v: (*int32)(nil), illegalPtr: false},
+		{v: (*int64)(nil), illegalPtr: false},
+		{v: (*uint)(nil), illegalPtr: false},
+		{v: (*uint8)(nil), illegalPtr: false},
+		{v: (*uint16)(nil), illegalPtr: false},
+		{v: (*uint32)(nil), illegalPtr: false},
+		{v: (*uint64)(nil), illegalPtr: false},
+		{v: (*float32)(nil), illegalPtr: false},
+		{v: (*float64)(nil), illegalPtr: false},
+		{v: (*string)(nil), illegalPtr: true},
+		{v: (*bool)(nil), illegalPtr: false},
 	}
 
-	for test := range tests {
-		NewDatatypeFromValue(test)
-		// Test again for usage with ptrs
-		NewDatatypeFromValue(&test)
```

Comment:
This would have been (but wasn't) testing `*interface{}` which we do not handle.